### ISSUE TITLE
Re#736 - Element Styles

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#738]
See master issue for instructions/guidelines for implementation: 
https://github.com/backdrop/backdrop-issues/issues/736

![style guide backdrop](https://cloud.githubusercontent.com/assets/5607236/6276238/6816d762-b853-11e4-84a2-e1a9375062ea.png)

**And later down the page...**
![style guide backdrop](https://cloud.githubusercontent.com/assets/5607236/6276428/bb327770-b854-11e4-882f-f95988c44ae4.png)

**And later there's more...**
![style guide backdrop](https://cloud.githubusercontent.com/assets/5607236/6276512/72d7c678-b855-11e4-90d4-4ca94b7f3164.png)
![style guide backdrop](https://cloud.githubusercontent.com/assets/5607236/6276531/96bb98d0-b855-11e4-901d-c167ddbdc2a8.png)

**Comment by jenlampton** [comment]
Since this issue is the one that will have the biggest impact in Backdrop 1.1 (since it will have the new font) I made some changes to the styleguide to unblock it.

**Comment by quicksketch** [comment]
This issue was inaccurately marked postponed. I tried out all the current PRs we have in a single local branch, including tabs, tables, pager, and messages. But overall it's hard to evaluate them accurately without the fonts being correct. Could we move forward on this one as soon as possible to minimize the amount of adjustment we'd need to make on other items?

**Comment by ronnienorwood** [comment]
@quicksketch @wesruv @jenlampton  I went ahead and made the changes then submitted a PR solely to change the font so that can be QAed.
Not sure what I am doing on the styleguide changes.  Am I modifying the css file on the styleguide module ??  If so, how do I differentiate between changes to the seven theme and any other theme ??  Hope that makes sense.

**Comment by serundeputy** [comment]
Hi @ronnienorwood  ; I think this has merge conflicts because `seven`'s css files have moved into a `css` directory.  Are you working against an older version of backdrop?  If so this could help: http://serundrop.staffordtavern.com/backdropcms-core-repo

cheers,
~Geoff

**Comment by quicksketch** [comment]
> Not sure what I am doing on the styleguide changes. Am I modifying the css file on the styleguide module ?? If so, how do I differentiate between changes to the seven theme and any other theme ??

To modify the styleguide at http://backdrop-ops.github.io/styleguide, make a checkout of https://github.com/backdrop-ops/styleguide. There is no PHP code in that repository. It's just a raw HTML file and CSS. Just modify the seven/style.css file to change the styleguide. However, this entire repository is just for reference. It's a sandbox to demonstrate the look we want to accomplish in Backdrop itself.

For the pull request (thanks for filing that!) We will want to include a local copy of the font Open Sans directly in the theme, rather than require an internet connection to Google Fonts.

**Comment by ronnienorwood** [comment]
@serundeputy  Not sure what happened because I forked into a whole new repo right before I made my changes then submitted the PR.  I just did the same again and now it looks fine, with the css directory there.

**Comment by ronnienorwood** [comment]
Hi @quicksketch,

> To modify the styleguide at http://backdrop-ops.github.io/styleguide, make a checkout of https://github.com/backdrop-ops/styleguide. There is no PHP code in that repository. It's just a raw HTML file and CSS. Just modify the seven/style.css file to change the styleguide. However, this entire repository is just for reference. It's a sandbox to demonstrate the look we want to accomplish in Backdrop itself.

I pretty much understood this.  That is what I did when I originally modified the styleguide.  Just not sure where to make the changes in BD itself or how to test them.  That is why I wondered (over in #736) if I should be modifying the styleguide module itself.

> For the pull request (thanks for filing that!) We will want to include a local copy of the font Open Sans directly in the theme, rather than require an internet connection to Google Fonts.

I'm not sure how to do this if I don't use the import statement in the seven.base.css file.  I can do some more research.  Is that what you are referring to ??

**Comment by quicksketch** [comment]
> I'm not sure how to do this if I don't use the import statement in the seven.base.css file. I can do some more research. Is that what you are referring to ??

Yes, instead of doing an @import to a CSS file hosted at Google, you can either do an @import to a local CSS file that points at the same fonts, or just include the font-face definitions directly in the seven.base.css file. You can select and download the font with the right font weights at http://www.google.com/fonts#UsePlace:use/Collection:Open+Sans (there's a icon in the upper right for downloading).

Once the font is downloaded, add the font files to Seven (probably in a new fonts directory within seven), then reference the font files the same way Google does in their hosted files:

http://fonts.googleapis.com/css?family=Open+Sans:400,300,600,700

**Comment by ronnienorwood** [comment]
Ok @quicksketch,

Thanks for the direction.  I downloaded the zip file with the fonts from Google and placed them in a fonts folder.  I copied the font-face defs directly into the seven.base.css file and modified the url on each.  The first one looks like this:

@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-weight: 300;
  src: local('Open Sans Light'), local('OpenSans-Light'), url(core/themes/seven/fonts/OpenSans-Light.ttf) format('ttf');
}

When I test it and look in firebug, it still doesn't appear to recognize the open sans font.  So that's where I am so far.

**Comment by quicksketch** [comment]
> src: local('Open Sans Light'), local('OpenSans-Light'), url(core/themes/seven/fonts/OpenSans-Light.ttf) format('ttf');

I think the issue is that within CSS files, paths are relative to the css file itself. So the path should probably be `url(../fonts/OpenSans-Light.ttf)`.

**Comment by ronnienorwood** [comment]
@quicksketch,

That didn't do the trick.  @wesruv do you want to take a stab at this ??

**Comment by wesruv** [comment]
@ronnienorwood sure thing!

**Comment by wesruv** [comment]
@dariusgarza maybe you could save me some brain power... what weights are we using with Open Sans?
We have the option of: 300, 400, 600, 700, 800
I'm hoping to chop out a few for load reasons, but if we want to use them all it's not that big of a deal.

**Comment by wesruv** [comment]
@ronnienorwood 
Here's a PR that just adds the font and sets it to the body element.
Did you want to take it the rest of the way? If you setup my backdrop repo as a remote you should be able to grab my branch/commits and finish it off.
If not that's cool, I can polish it off.

**Comment by ronnienorwood** [comment]
@wesruv

> Here's a PR that just adds the font and sets it to the body element.

where can I see the PR ??

**Comment by wesruv** [comment]
@ronnienorwood 
https://github.com/backdrop/backdrop/pull/785

**Comment by quicksketch** [comment]
I merged in https://github.com/backdrop/backdrop/pull/785 into the new `1.x-new-style` branch. It looks like we still need to do additional work here to do all the element styles themselves, no? Headings, paragraph margins, etc.

**Comment by ronnienorwood** [comment]
@wesruv
Good idea to create a separate file.  I didn't know you had to include all of those files in the fonts folder.  The only ones I had were the .ttf ones that came in the zip file I downloaded from google fonts.  Where did all the others come from ??
@quicksketch
I made a folder for the latest repo to use for testing and pulled in the 1.x-new-style changes.  I added the styleguide module, enabled it, and updated the permission, but the styleguide option doesn't show up in the appearance menu like it did in a previous version I had.  Not sure if I'm missing something.

**Comment by wesruv** [comment]
@ronnienorwood 
I kept trying to get the source files from Google, but that was a total pain. I ended up going to Font Squirrel. Their `@font-face` CSS wasn't what we wanted, but everything else was spot on.

Might try clearing cache to see if it makes the menu item show up, or just navigate to `admin/appearance/styleguide/seven`

**Comment by ronnienorwood** [comment]
@wesruv

> I kept trying to get the source files from Google, but that was a total pain. I ended up going to Font Squirrel. Their @font-face CSS wasn't what we wanted, but everything else was spot on.

Good info to know.

> Might try clearing cache to see if it makes the menu item show up, or just navigate to admin/appearance/styleguide/seven

Clearing cache didn't fix it and when I navigate to admin/appearance/styleguide/seven on my latest repo, it just goes back to the main appearance page, like it isn't recognizing that the module is enabled but it is.  Have to try something else.

**Comment by ronnienorwood** [comment]
@wesruv
@quicksketch
Disabling, uninstalling, and then reinstalling fixed the styleguide module problem but when I mouse over "Open Sans" in firebug, it still just looks the same as the lucida grande font.

**Comment by wesruv** [comment]
@ronnienorwood do any chat or screen sharing that we can collab on? e.g. hangouts, skype, etc?

**Comment by ronnienorwood** [comment]
@wesruv
I use skype.  I'll be on for another half hour or so now.  Otherwise, I'll be back on in the morning.  ronnie.norwood.33 is my handle.

**Comment by wesruv** [comment]
PR to get body font size in to fix admin tabs issue:
https://github.com/backdrop/backdrop/pull/797

**Comment by wesruv** [comment]
@ronnienorwood how are the styles for the other elements coming along?

**Comment by ronnienorwood** [comment]
@wesruv
I worked on them some this morning.  I'm hoping to spend some more time on it this weekend and get it hammered out.

**Comment by wesruv** [comment]
Excellent :+1:

**Comment by ronnienorwood** [comment]
@wesruv @quicksketch @jenlampton 
I'm about ready to create a PR for this.
Your feedback on a few issues I ran into.
The highlighted words in the emphasis and italic sentences are not slanted like the master styleguide.  I thought that since they were html tags, they would automatically be slanted.
The master styleguide has the quotes tag set to none so there are no quotation marks in the sentence.  I'm not sure why it was done that way.
On the master styleguide, the strong tag has a different color than the b tag so I just changed my strong tag color to match my b tag color.
Finally, on the h1 heading tag, I set font-weight to 600 to match the setting on the master styleguide but my letters look thicker than the ones on the master styleguide.

**Comment by wesruv** [comment]
@ronnienorwood is there a place we can see the code? Did you push it to your fork?

**Comment by wesruv** [comment]
@ronnienorwood I've also noticed that our bold doesn't match some of the style guide's bold. I have more weights in the repo, but didn't set all of them up. I can add semi-bold to the available font weights (right now we only have bold).

**Comment by ronnienorwood** [comment]
@wesruv 

> is there a place we can see the code? Did you push it to your fork?

I pushed and committed it but I'm not seeing my changes in my repo.

> I've also noticed that our bold doesn't match some of the style guide's bold. I have more weights in the repo, but didn't set all of them up. I can add semi-bold to the available font weights (right now we only have bold).

That would be great.

**Comment by wesruv** [comment]
@ronnienorwood I'm not seeing it either? I flipped around your fork and it's branches and didn't see any commits from you.

Just to be sure, technically you should commit, then push :smile: 

What's the name of the branch you're working on? 
What commands are you typing when you push? 

Being more specific by saying `git push BRANCHNAME` (BRANCHNAME would be replaced with your branch's name) could help?

**Comment by ronnienorwood** [comment]
@wesruv 

> Just to be sure, technically you should commit, then push :smile: 

Doh!  I did it backwards.

**Comment by wesruv** [comment]
> The highlighted words in the emphasis and italic sentences are not slanted like the master styleguide. I thought that since they were html tags, they would automatically be slanted.
Looks like that's also an issue with how the font was included with @font-face, I'll fix it up.

>The master styleguide has the quotes tag set to none so there are no quotation marks in the sentence. I'm not sure why it was done that way.
I think that may have just been an oversight? @dariusgarza:
Here's default state:
![style guide backdrop](https://cloud.githubusercontent.com/assets/5607236/6908238/acd6290a-d709-11e4-87bd-b50d45b830d7.png)

Here's what's in the styleguide:
![style guide l backdropdev com-1](https://cloud.githubusercontent.com/assets/5607236/6908241/b03ebb66-d709-11e4-8b08-0a6454b04e36.png)

> On the master styleguide, the strong tag has a different color than the b tag so I just changed my strong tag color to match my b tag color.
> Finally, on the h1 heading tag, I set font-weight to 600 to match the setting on the master styleguide but my letters look thicker than the ones on the master styleguide.
These two are related to the italic issue that I can get to.

**Comment by wesruv** [comment]
@ronnienorwood 
Forgot to CC you above, I also noticed that your branch is a little behind 1.x-new-style, you'll want to update that.

Can you also add the page bg color? You can see in the above screenshots that the styleguide has a slightly gray bg. That's not specifically covered in any of the tickets, I think it's most appropriate to tackle it in this one.

I'm going to try to get to the bold/italic issues by tomorrow night.

**Comment by ronnienorwood** [comment]
@wesruv 

> I also noticed that your branch is a little behind 1.x-new-style, you'll want to update that.

Is there a command for that ??

> Can you also add the page bg color?

Background color change committed.

**Comment by wesruv** [comment]
@ronnienorwood not that I'm aware of :/

I've been doing stuff like this:
https://help.github.com/articles/syncing-a-fork/

If you don't have a remote called upstream you can do this:
`git remote add upstream git@github.com:backdrop/backdrop.git`

Which should get you setup.

**Comment by jenlampton** [comment]
Since getting the fonts in is blocking our progress on all the other admin theme / styleguide issues, I'm going to split the fonts out into their own issue, here: https://github.com/backdrop/backdrop-issues/issues/875

**Comment by jenlampton** [comment]
@ronnienorwood I've un-assigned you from this issue since it's been 20 days since you've worked on it last.  @Graham-72 has some time and might be able to help us wrap this up.

**Comment by ronnienorwood** [comment]
Hi @jenlampton,

> Can you also add the page bg color? You can see in the above screenshots that the styleguide has a slightly gray bg. That's not specifically covered in any of the tickets, I think it's most appropriate to tackle it in this one.
I'm going to try to get to the bold/italic issues by tomorrow night.

@wesruv was going to add more bold & italic entries to the fonts file so I was waiting on that to finish this out.

The last thing I had done was change the page bg color.

**Comment by Graham-72** [comment]
I have been reviewing the styles in admin and the style guide today and identified some changes to branch 1.x-new-style. I have not yet made a PR for more changes to styles.css because I had better wait for my PR for issue #753 Form Elements to be checked.

**Comment by wesruv** [comment]
@ronnienorwood sorry it took me so long, I got crazy busy. Just got a PR in to fix the italics here: https://github.com/backdrop/backdrop/pull/825

Once it's merged into `1.x-new-style` you can update and keep moving. We're shooting to get these admin theme tickets **done** by the end of the month. If you don't have time between now and then push up what you have and I'll finish it off, thanks so much for your help and patience!

**Comment by Graham-72** [comment]
I have made a new PR at https://github.com/backdrop/backdrop/pull/831
to make changes to one of the stylesheets in 'seven'
and modify the following in accordance with styleguide:
    background color
    para line height, list line height,
    blockquote

**Comment by quicksketch** [comment]
Thanks @Graham-72! This looks pretty good to me. I tested it out and I'm glad to see the background color getting set.

It looks like this is going a bit beyond "base" styles in some places, such as:

```css
+ul,
+.block ul,
+.item-list ul{
```

Should the base.css file be kept to just element styles? Or are overridding core-provided styles like this allowed? @wesruv and @jenlampton would have a better idea then myself.

**Comment by ronnienorwood** [comment]
@wesruv,

Looks like @Graham-72 is finishing this up now.

**Comment by wesruv** [comment]
@quicksketch @Graham-72 
Ugh. When I look at the name of the file, I agree with quicksketch, when I think about being DRY, I like Graham-72's solution better. I loathe that core is styling `.block ul` and `.item-list ul` because it makes things awkward downstream.

I vote to keep things DRYer and go with what @Graham-72 Graham has there. That said if we go that way we should get rid of this declaration block in style.css:17
```
.block ul,
.item-list ul {
  list-style-type: disc;
  list-style-image: none;
  margin: 0.25em 0 0.25em 1.5em; /* LTR */
}
```

**Comment by quicksketch** [comment]
Sounds good to me @wesruv. I was leaning towards keeping it base.css too. @Graham-72, could you remove those lines from style.css then? We also need to make sure the RTL styles are removed and maintained in the base.css file adequately. I'll add a comment to the current PR.

**Comment by Graham-72** [comment]
@quicksketch - I have modified my PR https://github.com/backdrop/backdrop/pull/831 and will make a new PR for removing lines from style.css

**Comment by Graham-72** [comment]
Something that doesn't seem to show up on the style guide page is a problem with line spacing of text within a vertical tab. Here is what it looks like at the moment - 
![basic page - backdrop starter 2015-04-23 07-57-00](https://cloud.githubusercontent.com/assets/10559274/7291935/7f914e6a-e98e-11e4-9b21-31633d8879fa.png)

I will gladly produce a PR to alter line height which is currently set at 1

**Comment by quicksketch** [comment]
I merged in https://github.com/backdrop/backdrop/issues/835 before I saw this last comment @Graham-72. Let's make a follow-up for styling vertical tabs.

I think we're done with this issue and all the parts in it. Is there anything not yet done that's listed in the original post?

**Comment by quicksketch** [comment]
> Let's make a follow-up for styling vertical tabs.

Actually I take that back. I'll cherry-pick the commit from https://github.com/Graham-72/backdrop/commit/09e640f31031d2fa0cb5746a04aad0a4c1289f53.

**Comment by quicksketch** [comment]
Okay merged in the Vertical Tabs change as well. Let's close this issue up and make new issues for remaining pieces. The number of PRs referencing this issue are getting a bit confusing.
